### ISSUE TITLE
Restructure Kestrel docs

### ISSUE THREAD

**Issue by JamesNK** [dotnet/AspNetCore.Docs#19986]
The docs page for using Kestrel has grown over time and is now too long - 69 minutes. And more content is coming for .NET 5. It needs to be split up and thought be put into how content is organized.

https://docs.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel

Mostly this work will be reorganizing existing content, but I think a new Overview page that introduces Kestrel and summarizes features and links to other pages would be needed.

I don't have a proposal for content layout yet.

**Comment by Rick-Anderson** [comment]
@JamesNK  when you have a proposed outline paste it here and I'll  try to implement it.

**Comment by JamesNK** [comment]
Here is a TOC that @Tratcher @halter73 and I came up with.

Most of the content exists today. Existing content could all use more work, but lets focus on splitting up what we have first. Future improvements can come later.

* Overview
  - [What is Kestrel](https://github.com/dotnet/AspNetCore.Docs/blame/e6558695ea23c04a4e0ad6e5cc9b13e6a4bd36ce/aspnetcore/fundamentals/servers/kestrel.md#L16-L31)
    - Features
  - Why use Kestrel (new)
  - Deploying Kestrel app - (new - related to https://github.com/dotnet/AspNetCore.Docs/issues/18411)
  - [Kestrel in ASP.NET Core apps (aka hello world)](https://github.com/dotnet/AspNetCore.Docs/blame/e6558695ea23c04a4e0ad6e5cc9b13e6a4bd36ce/aspnetcore/fundamentals/servers/kestrel.md#L79-L101)
  - Summarize other pages (new)
* [Endpoint Bindings](https://github.com/dotnet/AspNetCore.Docs/blame/e6558695ea23c04a4e0ad6e5cc9b13e6a4bd36ce/aspnetcore/fundamentals/servers/kestrel.md#L360-L868) (Connection Logging is removed)
  - Launchsettings.json and precedence (future)
  - Code vs appsettings.json config
  - [URL prefixes](https://github.com/dotnet/AspNetCore.Docs/blame/e6558695ea23c04a4e0ad6e5cc9b13e6a4bd36ce/aspnetcore/fundamentals/servers/kestrel.md#L901-L943) moves here
* [Configure Server Options](https://github.com/dotnet/AspNetCore.Docs/blame/e6558695ea23c04a4e0ad6e5cc9b13e6a4bd36ce/aspnetcore/fundamentals/servers/kestrel.md#L103-L358)
  - Code vs appsettings.json config
* [Transport configuration](https://github.com/dotnet/AspNetCore.Docs/blame/e6558695ea23c04a4e0ad6e5cc9b13e6a4bd36ce/aspnetcore/fundamentals/servers/kestrel.md#L870-L899)
  - #ifdef Libuv docs or warn that obsolete in 5
* Protocol support
  - Overview of supported HTTP protocols (new)
  - HTTP/1.1
    - [Request draining](https://github.com/dotnet/AspNetCore.Docs/blame/e6558695ea23c04a4e0ad6e5cc9b13e6a4bd36ce/aspnetcore/fundamentals/servers/kestrel.md#L2711-L2739)
  - HTTP/2
    - [Requirements](https://github.com/dotnet/AspNetCore.Docs/blame/e6558695ea23c04a4e0ad6e5cc9b13e6a4bd36ce/aspnetcore/fundamentals/servers/kestrel.md#L33-L47)
  - HTTP/3 (future) @jkotalik
    - Requirements (future) @jkotalik
* Logging and diagnostics
  - Configuring Kestrel logging (future)
  - Move [Connection logging](https://github.com/dotnet/AspNetCore.Docs/blame/e6558695ea23c04a4e0ad6e5cc9b13e6a4bd36ce/aspnetcore/fundamentals/servers/kestrel.md#L632-L644) here
  - Event counters (future
* [Host filtering](https://github.com/dotnet/AspNetCore.Docs/blame/e6558695ea23c04a4e0ad6e5cc9b13e6a4bd36ce/aspnetcore/fundamentals/servers/kestrel.md#L945-L985)
* [When to use Kestrel with a reverse proxy](https://github.com/dotnet/AspNetCore.Docs/blame/e6558695ea23c04a4e0ad6e5cc9b13e6a4bd36ce/aspnetcore/fundamentals/servers/kestrel.md#L51-L77)
  - Talk about why you might want it
  - Performance implications (future)
* Troubleshoot (future)
  - Client disconnection (future)
  - Common exceptions (future)

// @davidfowl

**Comment by Rick-Anderson** [comment]
> Why use Kestrel (new)

It’s true that in-proc means you no longer need to use Kestrel when running in IIS, but there are plenty of other scenarios for using Kestrel directly.
A)	Performance
B)	Quicker feature updates (e.g. gRPC)
C)	More customizable
D)	Cross platform (it’s the only AspNetCore server on Linux)

**Comment by JamesNK** [comment]
> > Why use Kestrel (new)
> 
> It’s true that in-proc means you no longer need to use Kestrel when running in IIS, but there are plenty of other scenarios for using Kestrel directly.
> A) Performance
> B) Quicker feature updates (e.g. gRPC)
> C) More customizable
> D) Cross platform (it’s the only AspNetCore server on Linux)

Yes, those are all good reasons to use Kestrel. Additionally, Kestrel has security hardening that makes it safe to use as an edge server (i.e. it can serve apps directly without requiring a proxy)

**Comment by Rick-Anderson** [comment]
cc @tdykstra  if you're interested.

**Comment by JamesNK** [comment]
Some of the Kestrel pages have been split out. However, there is lots in the proposed TOC that hasn't been done.
